swagger/src/main/scala/org/http4s/rho/swagger/TypeBuilder.scala

```diff
@@ -312,21 +318,26 @@ object TypeBuilder {
         if (t.typeArgs.nonEmpty) GenArray(fromType(t.typeArgs(1)))
         else GenArray()
       } else if (klass <:< typeOf[AnyVal]) {
-        fromType(
-          klass.members
-            .filter(_.isConstructor)
-            .flatMap(_.asMethod.paramLists.flatten)
-            .head
-            .typeSignature
-        )
+        val klassSymbolOption = klass.members
+          .filter(_.isConstructor)
+          .flatMap(_.asMethod.paramLists.flatten)
+          .headOption
+        klassSymbolOption match {
+          case Some(symbol) => fromType(symbol.typeSignature)
+          case None => fallBackDataTypeFromName(t)
+        }
       } else if (isObjectEnum(klass.typeSymbol)) {
         EnumDataType(klass.typeSymbol.asClass.knownDirectSubclasses.map(_.name.toString))
       } else {
-        val stt = if (t.isOption) t.typeArgs.head else t
-        ComplexDataType("string", qualifiedName = Option(stt.fullName))
+        fallBackDataTypeFromName(t)
       }
     }
 
+    private def fallBackDataTypeFromName(t: Type)(implicit st: ShowType): DataType = {
```

Comment:
I propose this name because this Swagger dataType is clearly a last result String that only displays the class name, which is far better than nothing, but should be plainly labeled as a last resort.